contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -879,16 +878,23 @@ contract LoanTokenLogicStandard is LoanTokenLogicStorage {
 		}
 	}
 
+	/**
+	 * @notice Check if entry price lies above a minimum
+	 *
+	 * @param loanTokenSent The amount of deposit.
+	 * @param collateralTokenAddress The token address of collateral.
+	 * @param minEntryPrice Value of loan token in collateral
+	 * */
 	function checkPriceDivergence(
-		uint256 leverageAmount,
 		uint256 loanTokenSent,
-		uint256 collateralTokenSent,
 		address collateralTokenAddress,
-		uint256 minReturn
+		uint256 minEntryPrice
 	) public view {
-		(, uint256 estimatedCollateral, ) =
-			getEstimatedMarginDetails(leverageAmount, loanTokenSent, collateralTokenSent, collateralTokenAddress);
-		require(estimatedCollateral >= minReturn, "coll too low");
+		/// @dev See how many collateralTokens we would get if exchanging this amount of loan tokens to collateral tokens.
+		uint256 collateralTokensReceived =
+			ProtocolLike(sovrynContractAddress).getSwapExpectedReturn(loanTokenAddress, collateralTokenAddress, loanTokenSent);
+		uint256 collateralTokenAmount = (collateralTokensReceived.mul(WEI_PRECISION)).div(loanTokenSent);
+		require(collateralTokenAmount >= minEntryPrice, "entry price above the minimum");
```

Comment:
it is more like `collateralTokenPrice` rather than `Amount` (misleading variable name)

Reply:
Modified the variable name to `collateralTokenPrice`